homework/homework2/src/App.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import Button from './components/button';
+import Input from './components/input';
+import Todo from './components/todo';
+function App () {
+  const [todos, setTodos] = useState([
+    {
+      id: 1,
+      description: 'Get out of bed',
+      deadline: '2017-09-11',
+      done: true,
+    },
+    {
+      id: 2,
+      description: 'Brush teeth',
+      deadline: '2017-09-10',
+      done: false,
+    },
+    {
+      id: 3,
+      description: 'Eat breakfast',
+      deadline: '2017-09-09',
+      done: false,
+    },
+  ]);
+
+  const [form, setForm] = useState({
+    id: todos.length+1,
+    description: '',
+    deadline: '',
+    done: false,
+  });
+
+  const handleChangeInput = event => {
+    const newTodo = event.target.value;
+    const key = [event.target.name];
+    const newForm = { ...form, [key]: newTodo };
+    setForm(newForm);
+  };
+
+  const handleDeleteTodo = (id) => {
+    const newTodos = todos.filter(item=>item.id!==id);
+    setTodos(newTodos);
+  };
+  const handleSubmit = event => {
+    event.preventDefault();
+    if (form.description&&form.deadline) {
+      setTodos([...todos, form]);
+      setForm({
+        id: todos.length+1,
+        description: '',
+        deadline: '',
+        done: false,
+      });
+    } else {window.alert('Please enter a todo');
+    }
+  };
+  const [showTodoList, setShowTodoList] = useState(false);
+  const [buttonText, setButtonText] = useState(false);
+
+  const toggleList = () => {
+    return setShowTodoList(!showTodoList), setButtonText(!buttonText);
+  };
+  return (
+      <div className="App">
+        <div className="App-header">
+          <img src={logo} className="App-logo" alt="logo" />
+          <h2>Welcome to React Todo App</h2>
+        </div>
+        <form className="form" onSubmit={handleSubmit}>
+          <Input
+            placeholder="description"
+            value={form.description}
+            name="description"
+            handleChangeInput={handleChangeInput}/>
+          <Input
+            placeholder="dd-mm-yyyy"
+            value={form.deadline}
+            name="deadline"
+            handleChangeInput={handleChangeInput}/>
```

Comment:
Better to use an input with type date here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

Reply:
I did